Standards/scs-xxxx-v1-provider-network-standard.md

```diff
@@ -0,0 +1,233 @@
+
+In practice, the great majority of OpenStack deployments will use Neutron with the ML2 core plugin and either the `router` or the `ovn-router` service plugins, which should support all extensions required by this standard.
+Because this standard tries to target the OpenStack API, rather than it's specific implementation, we might consider standardizing a minimal set of Networking API extensions that CSPs must provide.
+
+On the other hand, the mandatory set of API extensions follows directly from the mandated features, e.g. the `external-net`, `provider`, and `router` extensions, which must all be available if an external provider network is required.
+So, just specifying mandatory features rather than a certain set of extensions may actually be preferable, as it removes redundancy and thus the potential for inconsistency.
+
+## Standard
+
+If CSPs offer public IP addresses to projects, they **MUST** provide a provider network to every project to allocate public addresses.
+This provider network will in the following be referred to as the _standard provider network_.
+To avoid ambiguity, the standard provider network **SHOULD** be the only provider network available to projects by default.
+
+The standard provider network **MUST** be an external network, allowing it to be used as external gateway by virtual routers.
+The standard provider network **MUST** have the `is_default` flag set to True, and it **MUST** be the only network to do so.
+The standard provider network **MAY** be a shared network, allowing direct attachment of virtual servers.
+If the standard provider network is a shared network, it **MUST** enable port security to prevent projects from interfering with each other.
+
+If CSPs offer public IP addresses at all, they **MUST** provide a shared subnet pool for the allocation of at least one public /64 IPv6 prefix per project.
+This subnet pool **MUST** have the `is_default` flag set to True, and it **MUST** be the only IPv6 subnet pool to do so.
+If CSPs offer public IP addresses, they **SHOULD** also offer public IPv4 addresses.
+If they do offer public IPv4 addresses, they **MUST** provide at least one public Floating IP per project, but **MAY** also provide a shared subnet pool for the allocation of public IPv4 prefixes to project networks.
+If CSPs offer a subnet pool for the allocation of public IPv4 prefixes, it **MUST** have the `is_default` flag set to True, and it **MUST** be the only IPv4 subnet pool to do so.
+
+CSPs **MUST** externally route any public IP addresses allocated from subnets of the standard provider network.
+CSPs **MUST** provide dynamic routing for all project-allocated public IP-prefixes via the standard provider network.
+
+By default, users **SHOULD** be prohibited by policy from creating Networking RBAC rules, to prevent the creation of faux provider networks.
```

Comment:
Can you add an implementation note that shows the necessary policy change?

Reply:
I clicked the link and unfortunately this is not good documented by OpenStack itself. It my be better to attach an implementataion note and not just write in block text but have the policy change presented in a clear readable way. Otherwise people might overread it.. I have something in mind like:

### Policy adjustment for network rbac

OpenStack itself allows all user, regardless of role to change the accessability of network resources (e.g. networks, routers, security groups, etc). This allows malicious users to present networking resources to other client as coming from the CSP. To forbid this the following policy adjustments SHOULD be implemented in the `policy.yaml` file:

```
"create_rbac_policy": "rule:admin_only"
```

Reply:
Alright, I put this in an implementation notes supplement now.